File: Assets/Smash Way/Scripts CS/Const/AnimSettings.cs
using UnityEngine;
using System.Collections;

public class AnimSettings : MonoBehaviour {


	public bool AnimNew = false;
	public AnimationClip AnimPlay;
	public GameObject ObjParent;
	public AudioClip Sound;

	private bool One = true;

	public void OnMouseDown(){
		Doing();
	}

	void Doing(){
		if(One){
			if(gameObject.GetComponent<Animation>()){gameObject.GetComponent<Animation>().Stop();}
			gameObject.GetComponent<AudioSource>().PlayOneShot(Sound);
			if(AnimNew){
				ObjParent.GetComponent<Animation>().Play(AnimPlay.name);
			}
			One=false;
		}
	}
}

File: Assets/Smash Way/Scripts JS/Const/AnimSettings.js
#pragma strict


var AnimNew:boolean=false;
var AnimPlay:AnimationClip;
var ObjParent:GameObject;
var Sound:AudioClip;

private var One:boolean=true;

function OnMouseDown(){
Doing();
}

function Doing(){
if(One){
if(gameObject.GetComponent.<Animation>()){gameObject.GetComponent.<Animation>().Stop();}
gameObject.GetComponent.<AudioSource>().PlayOneShot(Sound);
if(AnimNew){
ObjParent.GetComponent.<Animation>().Play(AnimPlay.name);
}
One=false;
}
}